### ISSUE TITLE
SC2149  indexing a hash using escaped variable no longer works

### ISSUE THREAD

**Issue by antofthy** [koalaman/shellcheck#2823]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2149
- My shellcheck version (`shellcheck --version` or "online"): 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/your/interpreter
your script here

```

#### Here's what THE PAGE currently says:

For associative arrays, the $ should be escaped to avoid accidental dereferencing:
```bash
declare -A array
index='$1'
array[$index]=42
echo "$(( array[$index] ))"    # bash: array: bad array subscript
echo "$(( array[\$index] ))"   # 42
```

#### Here's what I wanted or expected to see:

The last two lines now gives...
```bash
> echo "$(( array[$index] ))"    # bash: array: bad array subscript
42
> echo "$(( array[\$index] ))"   # 42
0
```
Here is what was put in memory -- which shows the assignment is now not what was expected
```bash
>  declare -p array
declare -A array=(["\$1"]="42" )
```
